cluster-autoscaler/core/scaledown/legacy/wrapper.go

```diff
@@ -89,49 +88,11 @@ func (p *ScaleDownWrapper) StartDeletion(empty, needDrain []*apiv1.Node, current
 func (p *ScaleDownWrapper) CheckStatus() scaledown.ActuationStatus {
 	// TODO: snapshot information from the tracker instead of keeping live
 	// updated object.
-	return &actuationStatus{
-		ndt: p.sd.nodeDeletionTracker,
-	}
+	return p.sd.nodeDeletionTracker
 }
 
 // ClearResultsNotNewerThan clears old node deletion results kept by the
 // Actuator.
 func (p *ScaleDownWrapper) ClearResultsNotNewerThan(t time.Time) {
-	// TODO: implement this once results are not cleared while being
-	// fetched.
-}
-
-type actuationStatus struct {
-	ndt *deletiontracker.NodeDeletionTracker
-}
-
-// DeletionsInProgress returns node names of currently deleted nodes.
-// Current implementation is not aware of the actual nodes names, so it returns
-// a fake node name instead.
-// TODO: Return real node names
-func (a *actuationStatus) DeletionsInProgress() []string {
-	if a.ndt.IsNonEmptyNodeDeleteInProgress() {
```

Comment:
I dislike option 1, because it's so very specific to legacy implementation. Option 2 works, though it feels a bit hacky.

I would consider option 3 - keep info about empty/non-empty deletions separate in Actuator and have separate EmptyDeletionsInProgress() and DeletionsInProgress() (names could use work). The legacy check would be just `if len(DeletionsInProgress) > 0`. I think it's more generic than IsNonEmptyNodeDeleteInProgress() in that it allows rate-limiting the number of ongoing drains (as opposed to rate-limiting just the number of drains started per loop). 

An argument for why I think that may be useful: I think the per-loop limit makes more sense with empty nodes where we expect the deletion to only take short time. With parallel drain we could allow longer gracefulTerminationSeconds, but if we allow it and not rate-limit ongoing drains, we may end up with a huge number of ongoing drain operations.

Reply:
I think rate-limiting outgoing calls should really be done by actuator, but since `StartDeletion` already accepts empty & non-empty node lists separately, I decided to return two lists from `DeletionsInProgress`. I think that makes the API slightly more consistent, while allowing to keep semantics of the legacy check unchanged.